### DISCUSSION TITLE
How to use magma-nvim with python modules?

### DISCUSSION THREAD

**Discussion by b-m-f** [nix-community/nixvim#702] [Q&A] [answered]
I have `magma-nvim` set up like this:

```
      magma-nvim = {
        enable = true;
        imageProvider = "kitty";
      };

```
But I can not get the plugin to detect any kernels beside `python3` and whenever I try to evaluate anything that depends on external modules (installed via home-manager and I also tried extraPackages) I get the following error:

![Screenshot from 2023-11-15 17-40-51](https://github.com/nix-community/nixvim/assets/2843450/1c2cdb55-0b72-4c74-b74d-d3e9bed603de)

What is the proper way to get everything working?

**Comment by b-m-f** [comment, marked as answer]
I was able to get this working by rebuilding the plugin in my local config and injecting the necessary libraries.
You can follow this guide https://github.com/NixOS/nixpkgs/blob/nixos-23.05/doc/languages-frameworks/vim.section.md#what-if-your-favourite-vim-plugin-isnt-already-packaged-what-if-your-favourite-vim-plugin-isnt-already-packaged

and the original build config is here: https://github.com/NixOS/nixpkgs/blob/9fb122519e9cd465d532f736a98c1e1eb541ef6f/pkgs/applications/editors/vim/plugins/overrides.nix#L673

Then you can simply use this as the package for the plugin with [plugins.magma-nvim.package](https://nix-community.github.io/nixvim/plugins/magma-nvim.html#pluginsmagma-nvimpackage) `= YOUR_PACKAGE`

**Reply by GaetanLepage** [reply]
Thank you for sharing your solution !

**Comment by GaetanLepage** [comment]
It might be worth it for you to try the brand new `extraPythonPackages` option.
